src/main/java/org/jabref/logic/util/io/RegExpBasedFileFinder.java

```diff
@@ -185,27 +201,14 @@ public static String expandBrackets(String bracketString, BibEntry entry, BibDat
         }
 
         // Last step: check if the given file can be found in this directory
-        String filePart = fileParts[fileParts.length - 1].replace("[extension]", EXT_MARKER);
-        String filenameToLookFor = expandBrackets(filePart, entry, null, keywordDelimiter).replaceAll(EXT_MARKER, extensionRegExp);
-
-        try {
-            final Pattern toMatch = Pattern.compile('^' + filenameToLookFor.replaceAll("\\\\\\\\", "\\\\") + '$',
-                    Pattern.CASE_INSENSITIVE);
-            BiPredicate<Path, BasicFileAttributes> matcher = (path, attributes) -> toMatch.matcher(path.getFileName().toString()).matches();
-            resultFiles.addAll(collectFilesWithMatcher(actualDirectory, matcher));
-        } catch (UncheckedIOException | PatternSyntaxException e) {
-            throw new IOException("Could not look for " + filenameToLookFor, e);
-        }
-
-        return resultFiles;
-    }
-
-    private List<Path> collectFilesWithMatcher(Path actualDirectory, BiPredicate<Path, BasicFileAttributes> matcher) {
+        Pattern toMatch = createFileNamePattern(fileParts, extensionRegExp, entry);
+        BiPredicate<Path, BasicFileAttributes> matcher = (path, attributes) -> toMatch.matcher(path.getFileName().toString()).matches();
         try (Stream<Path> pathStream = Files.find(actualDirectory, 1, matcher, FileVisitOption.FOLLOW_LINKS)) {
-            return pathStream.collect(Collectors.toList());
-        } catch (UncheckedIOException | IOException ioe) {
-            return Collections.emptyList();
+            resultFiles.addAll(pathStream.collect(Collectors.toList()));
+        } catch (UncheckedIOException uncheckedIOException) {
```

Comment:
Why is this even an UncheckedIOException? 
If needed, the UncheckedException is just a wrapper for the IOEception. So you could call throw uncheckedException.getCause()

Reply:
Frankly, I am not sure why an `UncheckedIOException` is caught in this part of the code. I don't have much experience with nio.* but my interpretation of the API is that the `UncheckedIOException` must be caught in the parts of the code that make use of the reference.
I don't know if there are any other potential issues with a lazily loaded file system walk. Based on DirectoryStream and Files.walk I'd guess it could be thrown if depth > 1 and there is a cycle, hence, not in this part of the code unless it is changed.